gateway/sensor.go

```diff
@@ -171,10 +181,59 @@ func (g *Gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 	return nil
 }
 
+func parseErrorCode(errCode int) string {
+	switch errCode {
+	case 1:
+		return "error setting the board config"
+	case 2:
+		return "error setting the radio frequency config for radio 0"
+	case 3:
+		return "error setting the radio frequency config for radio 1"
+	case 4:
+		return "error setting the intermediate frequency chain config"
+	case 5:
+		return "error starting the gateway"
+	default:
+		return "unknown error"
+	}
+}
+
+// captureOutput creates a goroutine to capture C's stdout and log to the module's logger.
+// This is necessary because the sx1302 library only uses printf to report errors.
+func (g *Gateway) captureOutput() error {
+	stdoutR, stdoutW, err := os.Pipe()
+	if err != nil {
+		return err
+	}
+	// Redirect C's stdout to the write end of the pipe
+	C.redirectToPipe(C.int(stdoutW.Fd()))
+
+	g.workers.Add(func(ctx context.Context) {
```

Comment:
[opt] rather than define a function here to add as a worker, can we just pass in this whole function? similar to how we [do it for gps](https://github.com/viam-modules/gps/blob/bfae1ce1fbeae842c949e9d4e5b6024be3329a4b/rtk/gpsrtk.go#L64)